### ISSUE TITLE
[Bug]:  KeyError in PapertrailController#index, key not found: "ControlledVocabularyTerm"

### ISSUE THREAD

**Issue by kleintom** [SpeciesFileGroup/taxonworks#3780]
### Steps to reproduce the bug

```bash
1. Visit an existing Controlled Vocabulary Term, controlled_vocabulary_terms/14
2. Click on Report --> Paper Trail

I see this locally in development but not on sandbox. I'll post more info below.
```

### Environment

Development (docker)

### Version

5f85129135ee4281e6b12295c79173f14b2c46a4

**Comment by kleintom** [comment]
`papertrail?object_id=14&object_type=ControlledVocabularyTerm` gives me a rails error page saying
`KeyError in PapertrailController#index` `key not found: "ControlledVocabularyTerm"`
Extracted source is
```
    #   ApplicationEnumeration.data_models.inject({}){|hsh, k| hsh.merge!(k.name => k)}.fetch(klass_name)
    # elsif Rails.env.production?
    ::DATA_MODELS.fetch(klass_name)
    # else
    #   raise TaxonWorks::Error, 'whitelist attempted in unknown environment'
    # end
```
So it's passing `ControlledVocabularyTerm` as `klass_name` and failing (I think).

If I go to the rails console I get:
```
irb(main):001:0> DATA_MODELS.fetch("ControlledVocabularyTerm")
(irb):1:in `fetch': key not found: "ControlledVocabularyTerm" (KeyError)
irb(main):002:0> reload!
Reloading...
irb(main):003:0> DATA_MODELS.fetch("ControlledVocabularyTerm")
=> ControlledVocabularyTerm(id: integer, type: string, name: string, definition: text, created_at: datetime, updated_at: datetime, created_by_id: integer, updated_by_id: integer, project_id: integer, uri: string, uri_relation: string, css_color: string, position: integer)
```
So the reload! seems to fix it in the console, but I can't get it to work from the browser.

I tried moving `include Shared::IsData` to the end of the include list in `controlled_vocabulary_term.rb` since is_data says that's what you're supposed to do, but it didn't seem to make a difference.

**Comment by mjy** [comment]
This has the smell of a zeitwerk bug.   DATA_MODELS is an initialized constant that is may be set before all our Data models are loaded.  

@kleintom We have a number of config/initializers/constants that we may need to configure to load at different points, or move to be related to the model itself.  

I experimented a little with lib/sti_preload.rb, using Roles as a use case (I got identical errors to what you are seeing in a different context), sensu the Rails guides, it didn't seem to make a difference.

In general, given zeitwerk, we need to try and remove 'require' and 'require_dependency' in the base models from what I understand.  Doing this in conjunction with balancing constant inits (in particular focusing on ApplicationEnumeration where needed) is perhaps the route forward.
